hangry_games/src/models/game.rs

```diff
@@ -51,9 +50,14 @@ impl Game {
 
     pub fn close_area(&mut self, area: &crate::models::Area) {
         let connection = &mut establish_connection();
-        self.closed_areas.push(area.id);
+
+        let mut binding: Vec<Option<i32>> = vec![];
+        let closed_areas = self.closed_areas.as_mut().unwrap_or(&mut binding);
+        closed_areas.push(Some(area.id));
+        let closed_areas = closed_areas.clone();
```

Comment:
Why do we clone here? It looks like we're creating a local `closed_areas` (L55) that creates a collapsed? set? of the global `closed_areas` and `binding` (which is empty?) 

And then we're adding an entry on L56, and then saving that to state on L60. Do I follow correctly? If so, why the clone instead of setting the local `closed_areas`? I would presume it's to avoid passing a pointer to game state, though I don't understand fully.